### PR TITLE
Add hourly spend model

### PR DESCRIPTION
**Pull request by NiallRees**
Provides hourly reporting granularity. Uses `metering_history` to determine the latest hour to provide data for.

### CODE REVIEW

File: models/daily_spend.sql [hunk 1]
```diff
@@ -1,422 +1,11 @@
...
+    convert_timezone('UTC', hour)::date as date, -- get UTC date to align with Snowflake billing
+    service,
+    storage_type,
+    warehouse_name,
+    database_name,
+    sum(spend) as spend,
+    sum(spend_net_cloud_services) as spend_net_cloud_services,
+    any_value(currency) as currency
+from {{ ref('hourly_spend') }}
+group by 1,2,3,4,5
```
Comment (ian-whitestone) [inline models/daily_spend.sql:11]: have we reconciled that aggregating off of hourly spend matches?

File: models/hourly_spend.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+version: 2
```
Comment (ian-whitestone) [inline models/hourly_spend.yml:1]: can you add a note about any nuances / assumptions / caveats? any values subject to change upon rebuild (i.e. cloud services spend, storage)?

File: models/daily_spend.yml [hunk 3]
```diff
@@ -4,8 +4,8 @@ models:
   - name: daily_spend
     description: Table of daily spend broken down by services as listed on the invoice, with additional granularity.
     columns:
-    - name: date
-      description: Date of spend in UTC timezone.
+    - name: hour
+      description: Hour of spend in UTC timezone.
```
Comment (ian-whitestone) [inline models/daily_spend.yml:8]: this should be date, right?

File: models/hourly_spend.sql [hunk 4]
```diff
@@ -0,0 +1,429 @@
...
+    group by 1, 2, 3
+    union all
+    select
+        date,
+        'Stage' as storage_type,
+        null as database_name,
+        sum(average_stage_bytes) / power(1024, 4) as storage_terabytes
+    from {{ ref('stg_stage_storage_usage_history') }}
+    group by 1, 2, 3
+),
```
Comment (ian-whitestone) [inline models/hourly_spend.sql:57]: do these tables have records for the current day?

if they don't, storage values will be present for the latest day with 0 spend and users will need to handle that accordingly in their queries (we should call out as a caveat in the model docs as well).  previously daily spend filtered all input datasets to only go up to the last complete day so this wasn't an issue.

File: models/hourly_spend.yml (hunk 2 above)
Comment (NiallRees) [reply models/hourly_spend.yml:1]: Nothing changes, existing tests still pass.

File: models/hourly_spend.sql (hunk 4 above)
Comment (NiallRees) [reply models/hourly_spend.sql:57]: They do.

**Pull request merged**